server/project.go

```diff
@@ -247,61 +246,44 @@ func (s *Server) registerProjectRoutes(g *echo.Group) {
 			return echo.NewHTTPError(http.StatusNotFound, fmt.Sprintf("VCS not found with ID: %d", repositoryCreate.VCSID))
 		}
 
-		repositoryCreate.WebhookURLHost = s.profile.ExternalURL
-		repositoryCreate.WebhookEndpointID = uuid.New().String()
-		secretToken, err := common.RandomString(gitlab.SecretTokenLength)
+		// For a particular VCS repo, all Bytebase projects share the same webhook.
+		repositories, err := s.store.FindRepository(ctx, &api.RepositoryFind{
+			WebURL: &repositoryCreate.WebURL,
+		})
 		if err != nil {
-			return echo.NewHTTPError(http.StatusInternalServerError, "Failed to generate random secret token for GitLab").SetInternal(err)
-		}
-		repositoryCreate.WebhookSecretToken = secretToken
-
-		// Create a new webhook and retrieve the created webhook ID
-		var webhookCreatePayload []byte
-		switch vcs.Type {
-		case vcsPlugin.GitLabSelfHost:
-			webhookCreate := gitlab.WebhookCreate{
-				URL:                   fmt.Sprintf("%s/%s/%s", s.profile.ExternalURL, gitlabWebhookPath, repositoryCreate.WebhookEndpointID),
-				SecretToken:           repositoryCreate.WebhookSecretToken,
-				PushEvents:            true,
-				EnableSSLVerification: false, // TODO(tianzhou): This is set to false, be lax to not enable_ssl_verification
-			}
-			webhookCreatePayload, err = json.Marshal(webhookCreate)
+			return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to find repository with web url: %s", repositoryCreate.WebURL)).SetInternal(err)
+		}
+
+		repositoryCreate.WebhookURLHost = s.profile.ExternalURL
+		// If we can find at least one repository with the same web url, we will use the same webhook instead of creating a new one.
+		if len(repositories) == 0 {
+			// We use workspace id as webhook endpoint id
+			settingName := api.SettingWorkspaceID
+			settings, err := s.store.FindSetting(ctx, &api.SettingFind{Name: &settingName})
```

Comment:
Let's add a workspace ID to server and reference it. Similar to https://sourcegraph.com/github.com/bytebase/bytebase/-/blob/server/server.go?L194:2&subtree=true

Reply:
Done.